packages/mobile/src/components/core/Hyperlink.tsx

```diff
@@ -68,47 +68,55 @@ export const Hyperlink = (props: HyperlinkProps) => {
    * android doesn't return the correct layout for nested text elements
    * */
   useEffect(() => {
-    Object.entries(linkRefs.current).forEach(([index, ref]) => {
-      ref.measureInWindow((x, y, width, height) => {
-        setLinkLayouts((linkLayouts) => ({
-          ...linkLayouts,
-          [index]: { x, y, width, height }
-        }))
-      })
+    Object.entries(links).forEach(([index]) => {
+      const linkRef = linkRefs[index]
+      if (linkRef) {
+        linkRef.measureInWindow((x, y, width, height) => {
+          setLinkLayouts((linkLayouts) => ({
```

Comment:
does multiple sets in a for loop like this get merged into a single one in the end? otherwise wondering if we do map and then a set after?